### ISSUE TITLE
Registry for custom SERVICE handlers

### ISSUE THREAD

**Issue by Aklakan** [SPARQL-Anything/sparql.anything#60]
Jena lacks a system for having multiple SERVICE handlers to coexist peacefully next to each other.
Currently custom implementations of OpExecutor are needed such as [FacadeXOpExecutor](https://github.com/SPARQL-Anything/sparql.anything/blob/982332ee1043f385192199b841e957472484d1a0/sparql.anything.engine/src/main/java/com/github/spiceh2020/sparql/anything/engine/FacadeXOpExecutor.java#L154) or [OpExecutorServiceOrFile](https://github.com/SmartDataAnalytics/jena-sparql-api/blob/6138a2e389608a4a755f0e8113b4a91e7a1071d3/jena-sparql-api-sparql-ext/src/main/java/org/aksw/jena_sparql_api/sparql/ext/fs/OpExecutorServiceOrFile.java#L17).

For example, I'd like to mix the features of facade-x with those provided by the [rdf-processing-toolkit](https://github.com/SmartDataAnalytics/RdfProcessingToolkit/tree/develop/doc#federating-to-sorted-bzip2-encoded-n-triples) (Disclaimer: I am the developer of it).

In order to allow integration of SERVICE extensions from different projects Jena would need to be extended with such an infrastructure and the existing systems then need to be adapted to it.

**Comment by enridaga** [comment]
I agree it would be useful to have such flexibility. Please keep us updated!

**Comment by Aklakan** [comment]
Hi, I finally got around to create a Jira issue and made a proposal for the SERVICE extension point: https://issues.apache.org/jira/browse/JENA-215

Looking forward to hearing your opinion :)

**Comment by enridaga** [comment]
Nice! I've seen this is now included in release 4.2.0. Discussion for migrating to Jena 4 opened at #94

**Comment by luigi-asprino** [comment]
Hi, 

Things have evolved since 2021, FacadeXOpExecutor provides its own handling of OpPropFunc and OpBGP (in addition to OpService). Therefore, it cannot be replaced by a SERVICE handler only.
Mixing the two strategies (implementing an OpExecutor or implementing a SERVICE handler) might be confusing.
Therefore, I would tend to keep the FacadeXOpExecutor as it is and close the issue.
WDYT?

**Comment by justin2004** [comment]
> FacadeXOpExecutor provides its own handling of OpPropFunc and OpBGP (in addition to OpService).

but aren't `OpPropFunc` and `OpBGP` only handled within an `OpService`?  that is, i thought we got rid of the ability to put an `fx:properties` triple pattern in a query outside of a `x-sparql-anything:` service clause.

**Comment by luigi-asprino** [comment]
No, they aren't, that's the problem. With #150 they can come without a service operation. Moreover, `fx:anySlot` and properties in the triple pattern have required a custom handling of the `OpBGP`.

**Comment by luigi-asprino** [comment]
We can close this issue as FX approach cannot be implemented as a SERVICE handler only.
